rust/arrow/src/compute/kernels/boolean.rs

```diff
@@ -53,13 +53,18 @@ where
         right_data.null_bitmap(),
         |a, b| a & b,
     )?;
-    let values = op(&left_data.buffers()[0], &right_data.buffers()[0])?;
+    let left_buffer = &left_data.buffers()[0];
+    let right_buffer = &right_data.buffers()[0];
+    let values = op(
+        &left_buffer.slice(left_data.offset() / 8),
```

Comment:
This part does not work yet in all cases since we also need to take the length of the array data into account, which might be less than the buffer length.